src/devices/Common/Iot/Device/Graphics/BitmapImage.cs

```diff
@@ -70,5 +147,83 @@ public virtual void Clear(Color color = default)
                 }
             }
         }
+
+        /// <summary>
+        /// Sets pixel at specific position
+        /// </summary>
+        /// <param name="x">X coordinate of the pixel</param>
+        /// <param name="y">Y coordinate of the pixel</param>
+        /// <param name="color">Color to set the pixel to</param>
+        /// <remarks>The use of <see cref="SetPixel"/> and <see cref="GetPixel"/> is usually slow. For fast image updates, grab the underlying
+        /// raw buffer by calling <see cref="AsByteSpan"/> or use the <see cref="GetDrawingApi"/> method and use high-level drawing functions.</remarks>
+        public abstract void SetPixel(int x, int y, Color color);
+
+        /// <summary>
+        /// Gets the color of the pixel at the given position
+        /// </summary>
+        /// <param name="x">X coordinate of the pixel</param>
+        /// <param name="y">Y coordinate of the pixel</param>
+        /// <returns>The color of the pixel</returns>
+        /// <remarks>The use of <see cref="SetPixel"/> and <see cref="GetPixel"/> is usually slow. For fast image updates, grab the underlying
+        /// raw buffer by calling <see cref="AsByteSpan"/> or use the <see cref="GetDrawingApi"/> method and use high-level drawing functions.</remarks>
+        public abstract Color GetPixel(int x, int y);
+
+        /// <summary>
+        /// Return the data pointer as a raw span of bytes
+        /// </summary>
+        /// <returns>A span of bytes</returns>
+        public abstract Span<byte> AsByteSpan();
+
+        /// <summary>
+        /// Return the data pointer as a raw span of integers. For 32bit images, this equals to one pixel per entry.
+        /// </summary>
+        /// <returns>A span of integers</returns>
+        public virtual Span<int> AsIntSpan()
+        {
+            return MemoryMarshal.Cast<byte, int>(AsByteSpan());
+        }
+
+        /// <summary>
+        /// Saves this bitmap to a file
+        /// </summary>
+        /// <param name="filename">The filename to save it to</param>
+        /// <param name="fileType">The filetype to use.</param>
+        /// <remarks>
+        /// Generally, the method is not checking that the filename extension matches the file type provided.
+        /// </remarks>
+        public virtual void SaveToFile(string filename, ImageFileType fileType)
+        {
+            using (var fs = new FileStream(filename, FileMode.CreateNew))
+            {
+                SaveToStream(fs, fileType);
+            }
+        }
+
+        /// <summary>
+        /// Save the image to a stream
+        /// </summary>
+        /// <param name="stream">The stream to save the data to</param>
+        /// <param name="format">The image format</param>
+        public abstract void SaveToStream(Stream stream, ImageFileType format);
```

Comment:
Consider `WriteToStream` or just `WriteTo` since Stream is an argument it's obvious we're writing to stream